eZ/Publish/SPI/Search/Content/IndexerGateway.php

```diff
@@ -40,6 +40,18 @@ public function getContentInSubtree(string $locationPath, int $iterationCount):
      */
     public function countContentInSubtree(string $locationPath): int;
 
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     *
+     * @return \Generator list of Content IDs for each iteration
+     */
+    public function getContentWithContentTypeIdentifier(string $contentTypeIdentifier, int $iterationCount): Generator;
```

Comment:
For an interface a wider allowed type should be ok. What we're interested in is to be able to iterate over the results, so we can relax the constraints.

Ensure that what we're returning are integers.

```suggestion
     * @return iterable<int> list of Content IDs for each iteration
     */
    public function getContentWithContentTypeIdentifier(string $contentTypeIdentifier, int $iterationCount): iterable;
```